en/SUMMARY.md

```diff
@@ -1,96 +1,113 @@
 # Cocos2d-x Programmers Guide v3.16
 
 ## What is Cocos2d-x
-
 - [About Cocos2d-x](./about.md)
 - [Why choose Cocos2d-x]()
-- [...]()
+- [How to learn]()
+- [Where to get help]()
+- [How to contribute]()
 
 ## Basic Cocos2d-x Concepts
-
 - [Main Components](./basic_concepts.md)
 - [Director]()
-- [Scene]()
+- [Scenes and the Scene graph]()
 - [Sprites]()
-- [...]()
+- [Actions]()
 
 ## Sprites
-
 - [What are Sprites](./sprites.md)
 - [Creating Sprites]()
-- [...]()
+- [Using Spritesheets]()
+- [SpriteFrame Cache]()
+- [Sprite Manipulation]()
+- [Polygon Sprite]()
 
 ## Actions
-
-- [By and To, what is the difference](./actions.md)
+- [What are Actions](./actions.md)
 - [Basic Actions]()
 - [Sequences]()
-- [...]()
+- [Sequence Internals, what is really happening]()
 
 ## Building and Transitioning Scenes
-
 - [What is a Scene](./scenes.md)
 - [Creating a Scene]()
-- [...]()
+- [Transitioning from Scene to Scene]()
 
 ## UI Components
-
- - [Widgets, oh, my!](./ui_components.md)
- - [Label]()
- - [Menu]()
- - [Buttons]()
- - [...]()
+- [Widgets, oh, my!](./ui_components.md)
+- [Label]()
+- [Menu]()
+- [Buttons]()
+- [Checkbox]()
+- [Loading Bar]()
+- [Slider]()
+- [TextField]()
 
 ## Other Node Types
-
 - [TileMap](./other_node_types.md)
 - [Particle System]()
 - [Parallax]()
 
 ## Event Dispatcher
-
 - [What is the EventDispatch mechanism](./event_dispatch.md)
 - [5 types of event listeners]()
 - [FixedPriority vs SceneGraphPriority]()
 - [Touch Events]()
-- [...]()
+- [Keyboard Events]()
+- [Accelerometor Events()
+- [Mouse Events]()
+- [Creating Custom Events]()
+- [Registering Events with the Dispatcher)[]
 
 ## 3D
-
 - [Terminology](./3d.md)
 - [Sprite3D]()
 - [Animation]()
-- [...]()
+- [Camera]()
+- [Cubemap Texture]()
+- [Skybox]()
+- [Lighting]()
+- [Terrain]()
+- [Tools for creating 3D Sprites]()
+- [Advanced 3D Topics]()
```

Comment:
They are existing content, right? We will not add new content for 3D.

Reply:
yes already existing content

Reply:
The only thing I will add to 3D is a few more code example for things like touch events

Reply:
got it